### PR TITLE
Tx Fee calculator

### PR DESCRIPTION
**Pull request by panleone**
Allow dynamic calculation of fees

### CODE REVIEW

**Comment by panleone** [comment]
rebased on master

File: src/transaction.rs [hunk 1]
```diff
@@ -288,9 +304,10 @@ pub async fn create_transaction_internal(
     network: Network,
 ) -> Result<JSTransaction, Box<dyn Error>> {
     let mut builder = Builder::new(network, block_height);
-    let (nullifiers, change) = match inputs {
-        Either::Left(notes) => choose_notes(&mut builder, &notes, extsk, amount)?,
-        Either::Right(utxos) => choose_utxos(&mut builder, &utxos, amount)?,
+    let (transparent_output_count, sapling_output_count)  = if  to_address.starts_with(network.hrp_sapling_payment_address()) {(0,2)} else {(1,1)};
+    let (nullifiers, change, fee) = match inputs {
+        Either::Left(notes) => choose_notes(&mut builder, &notes, extsk, amount, transparent_output_count, sapling_output_count)?, //FOR THE MOMENT WE ONLY HAVE SHIELD TX WITH 2 OUTPUTS
```
Comment (Duddino) [inline src/transaction.rs:309]: Decapitalize the comment please

**Review by Duddino — APPROVED** [review]
tACK, properly calculated the fee of this tx:
https://testnet.rockdev.org/tx/4d56adc6703af2964ea830e97a9b58a0b3947de4b0447cec78289424ed73f58b

File: src/transaction.rs (hunk 1 above)
Comment (panleone) [reply src/transaction.rs]: done and merging

**Pull request merged**